bayesian/utility/v2/component_analyses.py

```diff
@@ -122,11 +124,14 @@ def ca_validate_input(input_json: Dict, ecosystem: str) -> Tuple[List[Dict], Lis
         if ecosystem == 'golang':
             _, clean_version = GolangDependencyTreeGenerator.clean_version(given_version)
             pseudo_version = gh.is_pseudo_version(clean_version)
+            # Strip module appended to the package name
+            package = package.split('@')[0]
 
         packages_list.append(
-            {"name": package, "version": clean_version, "given_version": given_version,
+            {"name": package, "given_name": given_package,
+             "version": clean_version, "given_version": given_version,
```

Comment:
Does given_name need to be passed into this? It does not seem to be used.

Reply:
This is used while building response. At line number 190, we are building a map with this value. This map is used by get_given_name_and_version() at line number 230, to extract given_name. The given_name is then passed to generate_recommendation() to actually build the response.

Reply:
I was referring to packages_list which only seems to be used in get_batch_ca_data. I do not see packages_list being used anywhere else.

Reply:
'packages_list' is existing flow and is used very much, its returned from this function to the caller (api_v2:227), which again pass this list to get response (api_v2:229).

Reply:
Does this function use any of the parameters that you are passing
https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/bayesian/api/api_v2.py#L229

Does this function use package_list
https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/bayesian/api/api_v2.py#L231

Reply:
Yes, both of this function use all parameters passed to it. Do you see otherwise?

Reply:
Please go ahead and merge, we can have this discussion offline. The code needs to refactored a lot.